### ISSUE TITLE
Cannot create Host on EC2.  Spins on "Installing Rancher Agent"

### ISSUE THREAD

**Issue by lwjameson** [rancher/rancher#7333]
I have used the UI multiple times to try and add a host to a newly created Master from EC2.  The process hangs at "Installing Rancher Agent".  This is what I see in the logs:

time="2017-01-06T18:55:31Z" level=info msg="stdout: Creating CA: /var/lib/cattle/machine/9d099016-8f03-4a33-a76b-c7d81f98a2f2/certs/ca.pem" resourceId: =1ph7
time="2017-01-06T18:55:31Z" level=info msg="stdout: Creating client certificate: /var/lib/cattle/machine/9d099016-8f03-4a33-a76b-c7d81f98a2f2/certs/cert.pem" resourceId: =1ph7
time="2017-01-06T18:55:31Z" level=info msg="stdout: Running pre-create checks..." resourceId: =1ph7
time="2017-01-06T18:55:31Z" level=info msg="stdout: Creating machine..." resourceId: =1ph7
time="2017-01-06T18:55:31Z" level=info msg="stdout: (hostx01) Launching instance..." resourceId: =1ph7
time="2017-01-06T18:55:46Z" level=info msg="stdout: Waiting for machine to be running, this may take a few minutes..." resourceId: =1ph7
time="2017-01-06T18:55:46Z" level=info msg="stdout: Detecting operating system of created instance..." resourceId: =1ph7
time="2017-01-06T18:55:46Z" level=info msg="stdout: Waiting for SSH to be available..." resourceId: =1ph7
time="2017-01-06T18:56:34Z" level=info msg="stdout: Detecting the provisioner..." resourceId: =1ph7
time="2017-01-06T18:56:34Z" level=info msg="stdout: Provisioning with ubuntu(systemd)..." resourceId: =1ph7
time="2017-01-06T18:57:16Z" level=info msg="stdout: Installing Docker..." resourceId: =1ph7
time="2017-01-06T18:58:56Z" level=info msg="stdout: Copying certs to the local machine directory..." resourceId: =1ph7
time="2017-01-06T18:58:57Z" level=info msg="stdout: Copying certs to the remote machine..." resourceId: =1ph7
time="2017-01-06T18:58:58Z" level=info msg="stdout: Setting Docker configuration on the remote daemon..." resourceId: =1ph7
time="2017-01-06T18:59:00Z" level=info msg="stdout: Checking connection to Docker..." resourceId: =1ph7
time="2017-01-06T18:59:00Z" level=info msg="stdout: Docker is up and running!" resourceId: =1ph7
time="2017-01-06T18:59:00Z" level=info msg="stdout: To see how to connect your Docker Client to the Docker Engine running on this virtual machine, run: docker-machine env hostx01" resourceId: =1ph7
time="2017-01-06T18:59:00Z" level=info msg="Machine Created" machineExternalId=9d099016-8f03-4a33-a76b-c7d81f98a2f2 resourceId=1ph7
time="2017-01-06T18:59:00Z" level=info msg="Creating and uploading extracted machine config" resourceId=1ph7
time="2017-01-06T18:59:00Z" level=info msg="Machine config file created and encoded." destFile="/var/lib/cattle/machine/9d099016-8f03-4a33-a76b-c7d81f98a2f2/hostx01.tar.gz" resourceId=1ph7
time="2017-01-06T18:59:00Z" level=info msg="Activating Machine" eventId=f216e108-a808-458d-a7c4-e179304aff05 resourceId=1ph7
time="2017-01-06T18:59:01Z" level=info msg="pulling rancher/agent:v1.1.3 image."
time="2017-01-06T18:59:12Z" level=info msg="Container created for machine" containerId=a97e356298822a60cedec92f67b188169f67b92c52cdb628e14576a72f74041c machineId=1ph7 resourceId=1ph7
time="2017-01-06T18:59:12Z" level=info msg="Rancher-agent for machine started" containerId=a97e356298822a60cedec92f67b188169f67b92c52cdb628e14576a72f74041c machineExternalId=9d099016-8f03-4a33-a76b-c7d81f98a2f2 resourceId=1ph7
time="2017-01-06T18:59:12Z" level=info msg="Creating and uploading extracted machine config" resourceId=1ph7
time="2017-01-06T18:59:12Z" level=info msg="Machine config file created and encoded." destFile="/var/lib/cattle/machine/9d099016-8f03-4a33-a76b-c7d81f98a2f2/hostx01.tar.gz" resourceId=1ph7

---
| Useful | Info |
| :-- | :-- |
|Versions|Rancher `v1.3.0` Cattle: `v0.175.2` UI: `v1.3.5` |
|Access|`github` `admin`|
|Orchestration|`Cattle`|
|Route|`host.containers`|

**Comment by lwjameson** [comment]
nvm...I needed to open another port on my server.

**Comment by onyxcodes** [comment]
Hi @lwjameson - can I please ask what your port/security config is that fixed your problem? I'm currently having this exact same issue and trying to find a fix. thanks!

**Comment by lwjameson** [comment]
Hi Onyx.  I needed to open port 8080 on the security group for the master server.  I had set it to only allow connections from myself.  Since I am mostly just evaluating Rancher at this point I opened it to everyone and am relying on the authentication on Rancher for security.  I am sure it can be set up more elegantly. One way or the other, the host machines in the cluster need access to 8080 to communicate with the rancher API.

Lorin
